src/sys/fuchsia/eventedfd.rs

```diff
@@ -0,0 +1,246 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use libc;
+use magenta;
+use magenta::HandleBase;
+use sys::fuchsia::{DontDrop, poll_opts_to_wait_async, sys};
+use std::mem;
+use std::os::unix::io::RawFd;
+use std::sync::{Arc, Mutex};
+
+/// Properties of an `EventedFd`'s current registration
+#[derive(Debug)]
+pub struct EventedFdRegistration {
+    pub token: Token,
+    pub (in sys::fuchsia) handle: DontDrop<magenta::Handle>,
+    pub rereg_signals: Option<(magenta::Signals, magenta::WaitAsyncOpts)>,
+}
+
+/// An event-ed file descriptor. The file descriptor is owned by this structure.
+#[derive(Debug)]
+pub struct EventedFdInner {
+    /// Properties of the current registration.
+    pub registration: Mutex<Option<EventedFdRegistration>>,
+
+    /// Owned file descriptor.
+    ///
+    /// Note that, while this is public, it should not be exported from the crate, as
+    /// `fd` is closed on `Drop`, so modifying `fd` is a memory-unsafe operation.
+    pub fd: RawFd,
```

Comment:
See the conversation [here](https://github.com/carllerche/mio/pull/629#discussion_r127229539). I kind of tend towards agreeing with you, since then you can determine the visibility from just looking at the declaration site, but it is unnecessary and more verbose.